pkg/splunk/enterprise/configuration.go

```diff
@@ -894,12 +895,15 @@ func getManualUpdateRefCount(client splcommon.ControllerClient, cr splcommon.Met
 func createOrUpdateAppUpdateConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject) (*corev1.ConfigMap, error) {
 	scopedLog := log.WithName("createOrUpdateAppUpdateConfigMap").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
 
-	crKindMap := make(map[string]string)
+	var crKindMap map[string]string
 	var configMapData, status string
 	var configMap *corev1.ConfigMap
 	var err error
 	var numOfObjects int
+	var mux sync.Mutex
 
+	mux.Lock()
```

Comment:
Can we move the following code to a new method which returns `configMap` and call them here & other places? that way we have a shorter block of code to protect, and should be faster too
```
mux.Lock()
defer mux.Unlock()
configMapName := GetSplunkManualAppUpdateConfigMapName(cr.GetNamespace())
namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: configMapName}
configMap, err = splctrl.GetConfigMap(client, namespacedName)
```

Reply:
This lock not only protects configMap but the crKindMap kind of things too